index.js

```diff
@@ -30,182 +117,82 @@ class ghostGoogleDrive extends StorageBase {
    * @returns {*}
    */
   save(file, targetDir) {
-    const _this = this;
-    return new Promise(function(resolve, reject) {
-      const key = _this.config.key;
-      const jwtClient = new google.auth.JWT(
-        key.client_email,
-        null,
-        key.private_key,
-        API_SCOPES,
-        null
-      );
-
-      jwtClient.authorize(function(err, tokens) {
-        if (err) {
-          console.log(err);
-          reject(err);
-          return;
-        }
-
-        const drive = google.drive({
-          version: API_VERSION,
-          auth: jwtClient
-        });
-        drive.files.insert(
-          {
-            resource: {
-              title: file.name,
-              mimeType: file.type
-            },
-            media: {
-              mimeType: file.type,
-              body: fs.createReadStream(file.path)
-            }
-          },
-          function(err, res) {
-            if (err) {
-              console.log(err);
-              reject(err);
-              return;
-            }
-            const { data } = res;
-            // make the url looks like a file
-            resolve("/content/images/" + data.id + "." + data.fileExtension);
-
-            drive.permissions.insert({
-              fileId: data.id,
-              supportsAllDrives: true,
-              supportsTeamDrives: true,
-              resource: {
-                  'type': 'anyone',
-                  'role': 'reader',
-              },
-              fields: 'id',
-            }, function(err, res) {
-              if (err) {
-                console.error(err);
-              } else {
-                console.log('Permission ID: ', res.id)
-              }
+    return new Promise((resolve, reject) => {
+      auth(this.config)
+        .then(client => {
+          upload(client, file)
+            .then(resp => {
+              console.log(resp);
+              resolve('/content/images/' + resp.data.id + '.' + resp.data.fileExtension);
+              setPermissions(client, resp.data);
```

Comment:
I think it's better to make sure `setPermissions` is complete before resolving from here, as it's also an API call (asynchronous). Also, how about using async/await?

Reply:
@robincsamuel I resolved the promise inside another .then() call, async/await would be out of style but we could refactor to use async/await in another PR. What are your thoughts?

Reply:
Yeah, that works. But can we just make the `setPermission` return a promise instead of a callback so that it's similar to other methods?

Reply:
Sounds good. Ill work on this when I get home

Reply:
Done